src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -300,22 +301,16 @@ class AppsResource @Inject() (
   private def maybePostEvent(req: HttpServletRequest, app: AppDefinition) =
     eventBus.publish(ApiPostEvent(req.getRemoteAddr, req.getRequestURI, app))
 
-  private def search(cmd: String, id: String): Iterable[AppDefinition] = {
-    /* Returns true iff `a` is a prefix of `b`, case-insensitively */
-    def isPrefix(a: String, b: String): Boolean =
-      b.toLowerCase contains a.toLowerCase
+  private def search(cmd: String, id: String, label: String): Iterable[AppDefinition] = {
+    /* Returns true if `a` is a prefix of `b`, case-insensitively */
+    def isPrefix(a: String, b: String): Boolean = b.toLowerCase contains a.toLowerCase
+    val selectors = if (label != null) Some(new LabelSelectorParsers().parsed(label)) else None
 
     service.listApps().filter { app =>
-      val appMatchesCmd =
-        cmd != null &&
-          cmd.nonEmpty &&
-          app.cmd.exists(isPrefix(cmd, _))
-
-      val appMatchesId =
-        id != null &&
-          id.nonEmpty && isPrefix(id, app.id.toString)
-
-      appMatchesCmd || appMatchesId
+      val appMatchesCmd = cmd == null || (cmd != null && cmd.nonEmpty && app.cmd.exists(isPrefix(cmd, _)))
```

Comment:
I just commented on this in a different version (?) of this patch, I just repeat the comments, so don't be confused by duplicates.

You do not have to perform the `cmd != null` checks. Generally, I would prefer if you converted values that are possible `null` to an `Option` first by using the `Option(...)` constructor.